### PR TITLE
Properly prioritize Maintainer-contributed icons

### PR DESCRIPTION
**Pull request by adamrusted**
PR serves to update our auto-assign workflow to prioritize maintainer-created `new icon` PRs as either priority 3 (if there is a linked issue) or priority 4 (if there is no issue).

### CODE REVIEW

File: .github/workflows/add-labels-priority.yml
Comment (mondeja) [inline .github/workflows/add-labels-priority.yml:73]: Quoting the [discussion](https://github.com/orgs/simple-icons/teams/maintainers/discussions/9):

> This is for all "[new icon](https://github.com/simple-icons/simple-icons/pulls?q=is%3Aopen+is%3Apr+label%3A%22new+icon%22)" PRs submitted by team members that fulfil a request **other than their own**.

Notice the part which indicates "other than their own". So the linked issue can't be an issue opened by the same user that has opened the PR. This is why it hasn't been added yet.

**Review by mondeja — CHANGES_REQUESTED** [review]

File: .github/workflows/add-labels-priority.yml
Comment (adamrusted) [reply .github/workflows/add-labels-priority.yml:73]: We're getting the linked issue above, can we pull the creator of that issue?

Comment (mondeja) [reply .github/workflows/add-labels-priority.yml:73]: Use the GH API and [jq](https://jqlang.github.io/jq/). But keep in mind that the linked issue can be multiple issues. In that case is returned by a comma separated string of numbers, so you'll need to iterate over them, that's the difficulty.

Comment (adamrusted) [reply .github/workflows/add-labels-priority.yml:73]: Either that, or much easier - we alter the definition of what warrants a Priority 3 / 4 to suit what _is_ capable with what we're able to do (linked to Issue = 3, not linked to Issue = 4). If I remember rightly, the Priority 4 stuff was always super niche and only us adding icons we wanted to see in the project ourselves!

Comment (mondeja) [reply .github/workflows/add-labels-priority.yml:73]: Looks reasonable to me. Feel free to edit it and update the Priority 3 project issues and PRs accordingly, then we'll merge this 👍🏼

Comment (adamrusted) [reply .github/workflows/add-labels-priority.yml:73]: We're not able to update the discussion now (it's been archived) - but I'm more than happy to log in our maintainer channels on the ✨shiny new discord✨

Comment (adamrusted) [reply .github/workflows/add-labels-priority.yml:73]: @mondeja - please see [this thread](https://discord.com/channels/1142044630909726760/1146108362543599646/1146108362543599646) in the Discord. Have updated the wording slightly - so would be good to get your approval on this before we push this PR.

Comment (mondeja) [reply .github/workflows/add-labels-priority.yml:73]: I'm fine with the "recency". Maybe we could just remove the Priority 4 moving them to 3, but I don't usually review icons, so I'm not sure if there would be too many PRs in that priority.

In the technical part, we can just ignore the change along time. It seems to me really overkilling to run a workflow daily just to move PRs between projects, but maybe in the future could be addressed.

**Review by mondeja — APPROVED** [review]
LGTM, feel free to merge it! ✔️

File: .github/workflows/add-labels-priority.yml
Comment (adamrusted) [reply .github/workflows/add-labels-priority.yml:73]: Would be happy to merge Priority 4 and 3, the likelihood of an icon falling under Priority 4 as-is is extremely low. I've just been using it as a holding area for the breaking changes in the new board, but we have the ability to filter them in/out of specific views, so not really necessary.

Comment (mondeja) [reply .github/workflows/add-labels-priority.yml:73]: As you prefer.

Comment (adamrusted) [reply .github/workflows/add-labels-priority.yml:73]: Let's merge as-is for now. We can deprecate Priority 4 at a later date.

**Pull request merged**
